
# De los requisitos a la recolección



## Objetivos

* Entender los requisitos de los datos
* Explorar las etapas de la recolección dedatos


## Contenido


1.  [Requisitos de los datos](#0)<br>
2.  [Recolección de datos](#2)<br>


<hr>


# Requerimientos de los datos <a id="0"></a>


<figure style = 'text-align:center'><img src="Datos/metodologia_req.jpg" width=350>
</figure>

&nbsp;El enfoque analítico elegido determina los requisitos de datos. Específicamente, los métodos analíticos que se utilizarán requieren cierto contenido de datos, formatos y representaciones, guiados por el conocimiento del dominio.
<br>
&nbsp;Concluimos que es posible automatizar el proceso de determinación de la cocina de una receta o plato determinado utilizando los ingredientes de la receta o el plato. Para construir un modelo, necesitamos datos extensos de diferentes cocinas y recetas.
<br>
&nbsp;La identificación de los datos requeridos cumple con la etapa de requisitos de datos de la metodología de ciencia de datos.
***

# Recolección de datos [data collection] <a id="2"></a>

<figure style = 'text-align:center'><img src="Datos/metodologia_recol.jpg" width=350></figure>
<br>
&nbsp;En la etapa inicial de recopilación de datos, los científicos de datos identifican y recopilan los recursos de datos disponibles. Estos pueden estar en forma de datos estructurados, no estructurados e incluso semiestructurados relevantes para el dominio del problema.
<br>

#### Web Scraping de Recetas de Comida en linea

Un investigador llamado Yong-Yeol Ahn extrajo decenas de miles de recetas de comida (cocinas e ingredientes) de tres sitios web diferentes, a saber:

<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig3_allrecipes.png" width=500>
<div align="center">
www.allrecipes.com
</div>
<br/><br/>
<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig4_epicurious.png" width=500>
<div align="center">
www.epicurious.com
</div>
<br/><br/>
<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig5_menupan.png" width=500>
<div align="center">
www.menupan.com
</div>
<br/><br/>


No necesitaremos realizar ninguna recopilación de datos, ya que los datos que necesitamos para cumplir con el objetivo definido en la etapa de comprensión de negocios están disponibles.
<br>

#### Leemos los datos y echemos un vistazo.

In [2]:
import pandas as pd
recipes = pd.read_csv("Datos/recipes.csv")

In [6]:
recipes.head()

country angelica anise anise_seed apple apple_brandy apricot armagnac  \
0  Vietnamese       No    No         No    No           No      No       No   
1  Vietnamese       No    No         No    No           No      No       No   
2  Vietnamese       No    No         No    No           No      No       No   
3  Vietnamese       No    No         No    No           No      No       No   
4  Vietnamese       No    No         No    No           No      No       No   

  artemisia artichoke  ... whiskey white_bread white_wine  \
0        No        No  ...      No          No         No   
1        No        No  ...      No          No         No   
2        No        No  ...      No          No         No   
3        No        No  ...      No          No         No   
4        No        No  ...      No          No         No   

  whole_grain_wheat_flour wine wood yam yeast yogurt zucchini  
0                      No   No   No  No    No     No       No  
1                      No   No   No  No    No     No       No  
2                      No   No   No  No    No     No       No  
3                      No   No   No  No    No     No       No  
4                      No   No   No  No    No     No       No  

[5 rows x 383 columns]

In [9]:
recipes.shape

(57691, 383)

<p>
Entonces, nuestro conjunto de datos consta de 57,691 recetas. Cada fila representa una receta, y para cada receta, se documenta la cocina correspondiente y si existen 384 ingredientes en la receta o no, comenzando con almendra (<i>almond</i>) y terminando con calabacín (<i>zucchini</i>).</p>

***


Ahora que la etapa de recopilación de datos está completa, los científicos de datos suelen utilizar estadísticas descriptivas y técnicas de visualización para comprender mejor los datos y familiarizarse con ellos. Los científicos de datos, esencialmente, exploran los datos para:

* entender su contenido,
* evaluar su calidad,
* descubrir cualquier idea preliminar interesante, y,
* determinar si se necesitan datos adicionales para llenar los vacíos en los datos.